src/stores/search.ts

```diff
@@ -42,6 +47,7 @@ export const isSearchTypeSupported = (
 
 export interface SearchState {
   searchType: SearchType
+  recentSearches: Ref<string[]>
```

Comment:
Ideally our state variables should be base types and not refs. Is that what you meant when you said "odd behavior I had to deal with here"?

Reply:
So pinia actually handles reactives in stage quite elegantly and recommends it in a number of places. For example: https://pinia.vuejs.org/cookbook/composables.html#ssr

The odd behavior I had to deal with here was specifically with the watcher inside of `useStorage` failing to be evoked on changes after navigating from the homepage to the search page. I'm still not exactly sure why that occurs and wasn't able to fix it in testing without that 'hack'